jvm/src/Language/Java.hs

```diff
@@ -186,14 +187,8 @@ instance Coercible () 'Void where
   unsafeUncoerce _ = ()
 
 -- | Get the Java class of an object or anything 'Coercible' to one.
-classOf
-  :: ( Coercible a ('Class sym)
-     , KnownSymbol sym
-     )
-  => a
-  -> Sing sym
--- Silence redundant constraint warning.
-classOf x = sing `const` coerce x
+classOf :: forall a sym. (Coercible a ('Class sym), KnownSymbol sym) => a -> JNI.String
+classOf _ = JNI.fromChars (symbolVal (Proxy :: Proxy sym))
```

Comment:
Unfortunately, that warning was removed from `-Wall` in 8.0.2. But you're right, `jvm` should still work on 8.0.2. Il reinstate the workaround.